packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3268,6 +3272,20 @@ class _RouteEntry extends RouteTransitionRecord {
       'This route cannot be marked for pop. Either a decision has already been '
       'made or it does not require an explicit decision on how to transition out.',
     );
+    // Remove state that prevents a pop, e.g. LocalHistoryEntry[s].
+    int attempt = 0;
+    while (route.willHandlePopInternally) {
```

Comment:
All the other solutions I can think of would require adding new API and break other people, or messing with Navigator/Route class.